lib/datadog/core/configuration/ext.rb

```diff
@@ -27,11 +27,16 @@ module Agent
           # env var has "trace" in it, but it really applies to all products
           ENV_DEFAULT_PORT = 'DD_TRACE_AGENT_PORT'
           ENV_DEFAULT_URL = 'DD_TRACE_AGENT_URL'
+          ENV_DEFAULT_USE_SSL = 'DD_TRACE_AGENT_USE_SSL'
+          ENV_DEFAULT_TIMEOUT_SECONDS = 'DD_TRACE_AGENT_TIMEOUT_SECONDS'
+          ENV_DEFAULT_UDS_PATH = 'DD_TRACE_AGENT_UDS_PATH'
```

Comment:
Out of curiosity, as these cross-language, or created by us?

Reply:
I was going to comment on something similar -- if these are cross language, no concerns at all.

But if they're new and specific to Ruby, I would perhaps hesitate to add them as they're subsets of `DD_TRACE_AGENT_URL`, which is already supported by all libraries.

In my view, part of the problem with the transport proc started was it wasn't possible to set these values via code, exactly because it wasn't possible to set `DD_TRACE_AGENT_URL` via code. If you wanted to, you could already set them via env vars.

Thus, I think the new options via code look reasonable, but it's weird to add the new env vars because:

a) It creates inconsistency between client libraries. These env vars look very generic, so I can imagine a customer copy-pasting their existing Ruby config to another service, and then get confused when it doesn't work. (This has happened in the past with other env vars -- customers open an issue reporting "hey I use DD_FOO with python/java etc but it doesn't work for ruby -- help?)

b) It creates more duplication of settings -- having `DD_TRACE_AGENT_URL` and `DD_TRACE_AGENT_UDS_PATH` means introducing precedence, etc. And as you can kinda tell from the `AgentSettingsResolver`, there's already quite a lot of complexity in all this business.

Note that all of this comment is about the env vars -- I think it's fine to have the new settings for code (e.g. I think it's ok to not have a 1:1 correspondence between every code setting and env var; it's definitely not new to our settings)

Reply:
@marcotc These are not cross-language, I made them up. Searching through github it looks like I should probably replace `DD_TRACE_AGENT_UDS_PATH` with `DD_TRACE_AGENT_UNIX_DOMAIN_SOCKET`. For timeout, Python uses `DD_TRACE_AGENT_TIMEOUT_SECONDS` but PHP uses `DD_TRACE_AGENT_TIMEOUT_VAL`. 

@ivoanjo I think your points are valid and I'm in favor of simplifying the resolver as much as possible. I also agree that having multiple ways to set options via env var is a bit weird. But I think we may need the SSL and timeout env vars.
- It is possible to set your host via the cross-language vars `DD_TRACE_AGENT_HOST` and `DD_TRACE_AGENT_PORT` for an http connection. If a user does this, shouldn't they also be able to set the use of SSL via env var? 
- You can't set timeout via `DD_TRACE_AGENT_URL`. I don't think we want to go down the path of query strings?
- The only justification for `DD_TRACE_AGENT_UDS_PATH` is consistency with the http functionality. I don't feel strongly, but if we do keep the timeout and ssl env vars, I don't see the incremental added complexity of a UDS path    var as a big deal. 

If we want to adopt a policy of "If you want to configure via ENV vars and you want to either use SSL for http connections or you want to use UDS then you must use `DD_TRACE_AGENT_URL`" I don't object. I would just want to run it by TEE first to make sure we aren't missing an edge case.

Reply:
> It is possible to set your host via the cross-language vars DD_TRACE_AGENT_HOST and DD_TRACE_AGENT_PORT for an http connection. If a user does this, shouldn't they also be able to set the use of SSL via env var?

It's a very good point! I think this is all the result of very organic evolution. Having said that, using ssl is a quite advanced configuration: Afaik the datadog agent doesn't have built-in support for serving ssl. So if you're using ssl, it already means you have some proxying http server in front the agent, which probably means you know what you're doing.

For that reason, I would kinda lean towards letting `DD_TRACE_AGENT_URL` take care of ssl (and uds).

---

As for timeout: one one side, we never actually got asked to support it via env, which probably suggests it's not often used. So I would kinda wait for someone to ask.

But, I think it's reasonable if you want to preeempt that and go ahead. In that case, I'd probably suggest following the python convention, just because the PHP one is awful (cmon -- every setting is a value! it's like suffixing things with `_ENV_VAR` lol).

Reply:
@ivoanjo I think that's reasonable. I removed `DD_TRACE_AGENT_UDS_PATH` and `DD_TRACE_AGENT_USE_SSL` but left `DD_TRACE_AGENT_TIMEOUT_SECONDS` as is.

Reply:
Thank you so much for the improvements, @ekump!